bin/tests/e2e.go

```diff
@@ -0,0 +1,155 @@
+/*
```

Comment:
`bin/tests/e2e.go` 
`bin`/ is not look like a right place for go files.

Reply:
Agreed. Might make sense to just have a mage package.